manifests/instance.pp

```diff
@@ -276,7 +277,11 @@
     refreshonly => true,
   }
 
-  $redis_version_real = pick(getvar_emptystring('redis_server_version'), $minimum_version)
+  if $package_ensure =~ /^[0-9]\.[0-9]/ {
```

Comment:
The regex matches (I did not look for the end of the regex, only start with [0-9]\.[0-9]). Anyway I've improved it a bit (/^[0-9]+\.[0-9]/) so it will work when redis get to version 10 :).

I've added a couple more of specs usecases.